tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -117,6 +118,24 @@ async def compute(self):
                                   py_typecheck.type_string(type(self._value))))
 
 
+class FederatingStrategy(abc.ABC):
+  """Base class for intrinsic strategies.
+
+  Defines the contract between FederatingStrategy implementations and
+  FederatingExecutor. Includes some basic functionality that should be common
+  to all intrinsic strategies, e.g. placement and mapping functions.
+  See concrete implementations for more details.
+  """
+
+  def __init__(self, federating_executor):
+    self.federating_executor = federating_executor
+
+  @classmethod
+  @abc.abstractmethod
+  def validate_executor_placements(cls, executor_placements):
```

Comment:
I don't think we need this. Seems like we should simply move this to the init function fo the strategy. No reason to check this in the construction of the executor?

Reply:
Actually this makes sense why you have this because in this iteration it's the executor that owns the child executors. I would either leave as is (and I will remove it in follow up) or just remove it and inline. IT doesn't matter much which you choose I think? WDYT?

Reply:
Yep, it seemed like this could be cut out with your change. I'm okay with leaving as is if you are!